src/Core/DynamicMetadata/DynamicMetadataInterface.php

```diff
@@ -10,7 +10,7 @@
     /**
      * @return string The name of the metadata
      */
-    public function getName();
+    public static function getName();
```

Comment:
The reason why is that sometimes we need to get the DynamicMetadata name, even without having/creating an object for it. Thus the static method.
Do you have any better option?